### PR TITLE
Fix err 500 on listing /api/v1/namespaces with browsable api enabled

### PR DESCRIPTION
**Pull request by jerabekjiri**
Issue: AAH-2733

~~This feels like a hacky fix. I'm not sure why django says the method is POST on visiting the endpoint if browsable api is enabled.~~
For some reason, django browsable api form make POST (create) request which triggers access policy (`is_namespace_owner`) just on visiting endpoint. 

With browsable api enabled: visiting endpoint "request.method" is "POST", "action" is "create", even though request.META["REQUEST_METHOD"] is "GET". 
 
(if `?format=json`, endpoint works correctly even basic user is logged in)
Still investigating.

### CODE REVIEW

**Comment by jerabekjiri** [comment]
@drodowic Is it okay to use the `PULP_CONTENT_ORIGIN` variable to enable the `CustomBrowsableAPIRenderer` renderer class only for galaxy.ansible.com (and also for dev and stage)? I don't think I have any other way to test this.

**Comment by drodowic** [comment]
> @drodowic Is it okay to use the `PULP_CONTENT_ORIGIN` variable to enable the `CustomBrowsableAPIRenderer` renderer class only for galaxy.ansible.com (and also for dev and stage)? I don't think I have any other way to test this.

PULP_CONTENT_ORIGIN is explicitly set in each environment for galaxy.ansible.com so yes I think it's unique enough to use in this case.

**Review by drodowic — APPROVED** [review]

File: galaxy_ng/app/dynaconf_hooks.py [hunk 1]
```diff
@@ -533,6 +534,16 @@ def configure_authentication_backends(settings: Dynaconf) -> Dict[str, Any]:
...
 
 
+def configure_renderers() -> Dict[str, Any]:
+    "Set default renderer classes."
+    return {
+        "REST_FRAMEWORK__DEFAULT_RENDERER_CLASSES": [
+            'rest_framework.renderers.JSONRenderer',
+            'galaxy_ng.app.renderers.CustomBrowsableAPIRenderer'
+        ]
+    }
```
Comment (rochacbruno) [inline galaxy_ng/app/dynaconf_hooks.py:544]: There is no need to use dynaconf_hooks to set default values without any conditional, the hooks are valuable when the return dict to be merged with settings can vary depending on a condition.

The reason to perform the conditional here is that the hooks file runs at the very late stage of settings building, if the conditional is performed inside the `settings.py` the state of `settings` objects is unfinished, for example, another plugin can set a variable to override it.

```suggestion
import re

def configure_renderers(settings) -> Dict[str, Any]:
    "Set renderer classes conditionaly."
    if re.search(
        r'galaxy(-dev|-stage)*.ansible.com', settings.get('CONTENT_ORIGIN', "")
    ):
        # overrride the `REST_FRAMEWORK__DEFAULT_RENDERER_CLASSES`
        return {
            "REST_FRAMEWORK__DEFAULT_RENDERER_CLASSES": [
                'galaxy_ng.app.renderers.CustomBrowsableAPIRenderer'
            ]
        }
        
    return {}   # do nothing
```

**Review by rochacbruno — CHANGES_REQUESTED** [review]

File: docker/etc/settings.py [hunk 2]
```diff
@@ -41,6 +42,12 @@
 
 REST_FRAMEWORK__DEFAULT_RENDERER_CLASSES = ['rest_framework.renderers.JSONRenderer']
 
+# add CustomBrowsableAPI only for community (galaxy.ansible.com, galaxy-stage, galaxy-dev)"
+if re.search(r'galaxy(-dev|-stage)*.ansible.com', os.environ.get('PULP_CONTENT_ORIGIN', "")):
+    REST_FRAMEWORK__DEFAULT_RENDERER_CLASSES.append(
+        'galaxy_ng.app.renderers.CustomBrowsableAPIRenderer'
+    )
+
```
[the remark this answers is not in the document]
Comment (jerabekjiri) [reply docker/etc/settings.py:50]: @rochacbruno 
>  also, docker/etc/settings.py is only used by cloud deployment is that what you indend?


I want `CustomBrowsableAPIRenderer` to be used only in `galaxy.ansible.com` (and in the dev env as well), not in CRC for now.

[the remark this answers is not in the document]
Comment (jerabekjiri) [reply docker/etc/settings.py:50]: @rochacbruno 
>  also, docker/etc/settings.py is only used by cloud deployment is that what you indend?


I want `CustomBrowsableAPIRenderer` to be used only in `galaxy.ansible.com` (and in the dev env as well), not in CRC for now.

**Comment by jctanner** [comment]
Does this problem exist on the devstack in a way that we could have reproduced it?

**Comment by jerabekjiri** [comment]
@jctanner I can probably add test with `?format=api` to test the behavior.

**Review by rochacbruno — APPROVED** [review]

**Pull request merged**